### ISSUE TITLE
Change direction of difference() and derivative()

### ISSUE THREAD

**Issue by tobiasdirksen** [influxdata/influxdb#16398]
The difference and the derivative functions are very useful. However, I really lack a option to change the calculation to go from lead row minus current row, instead of current row minus lag row.

**Comment by russorat** [comment]
@tobiasdirksen thanks for opening this. 

@nathanielc do you think this should go in the Flux repo?

**Comment by russorat** [comment]
@tobiasdirksen i know it's been a while, but i'm curious if you found a way around this or if you could provide more details about your use case here.

**Comment by tobiasdirksen** [comment]
Thanks, I haven't really found a good solution yet. Say, we have a water flow meter with a volume flow counter in m³ (or any meter with a counter), and we want to calculate the volume consumption (m³) on daily sampled data. If we want the volume consumption on the 2020-08-04, that is naturally the difference between the volume flow counter at 2020-08-05 and 2020-08-04, thus lead row minus current row. I guess it is it a mater of whether you want to label the data with the start time or the end time, but at least with daily sampled date, you will always prefer to use the start time, in my opinion.

**Comment by sanderson** [comment]
I agree that it would be nice to specify which direction to perform the calculation in, but there is a workaround. You can sort the data by time (descending), execute `difference()` or `derivative()`, then sort by time (ascending). Sorts can be expensive, so this query won't be efficient, but it will give you the results you're looking for.

```js
data
  |> sort(columns: ["_time"], desc: true)
  |> difference()
  |> sort(columns: ["_time"])
```

**Comment by tobiasdirksen** [comment]
Sure, that's a workaround! But expensive as you said.
